### PR TITLE
Add a Lua/Lmod modulefile for Chapel on XC and EX systems.

### PR DESCRIPTION
**Pull request by gbtitus**
(Duplicate PR #16718 from 1.23-for-EX to master.)

(Original PR co-developed (python) and reviewed by @ronawho.)

The Programming Environment on HPE Cray EX and XC systems has added a
Lua/Lmod hierarchical module system to the existing flat Tcl-based one.
Here, add a Lua modulefile for Chapel to work with that.

(The original PR only targeted EX systems in order to to match up with
their delivery schedules.  Now that it's in the main line of development
it will target XC systems as well.)

While here, this also expands the number of configurations built into
the EX module.

A related change is that instead of using shell syntax to parameterize
here-documents to produce the modulefiles, this starts with standalone
template files and parameterizes those using a general-purpose python
script. This seems a much cleaner and more readable technique.

This addresses https://github.com/Cray/chapel-private/issues/1622 but does not fully resolve it.
This addresses https://github.com/Cray/chapel-private/issues/1619 but does not fully resolve it.

### CODE REVIEW

**Comment by gbtitus** [comment]
@ronawho just an FYI, since you reviewed all this when it was first merged into 1.23-for-EX. I'll wait for the pre-merge checks to run but don't need an actual review.

I've installed a built RPM from this into an EX UAI and used it to build `hello` and run that. And although I can't install an RPM on an XC system, I've confirmed that all the same files, plus and minus the expected added and removed ones respectively, are in the XC module. I've also cpio'ed the contents out of the XC RPM and used that as a `CHPL_HOME` and run `hello` on our XC build system.

File: doc/rst/platforms/cray.rst [hunk 1]
```diff
@@ -82,27 +99,28 @@ installed, do the following:
 
 
 Currently the number of Chapel configurations available on
-HPE Cray EX systems is quite limited.  Only the following have been built
+HPE Cray EX systems is somewhat limited.  Only the following have been built
```
Comment (lydia-duncan) [inline doc/rst/platforms/cray.rst:102]: Noticed when looking at a different documentation change, but I think this should be "HPE Cray EX systems *are* somewhat limited"

Comment (gbtitus) [reply doc/rst/platforms/cray.rst:102]: Not that I'm an English major, but no, I think the verb needs to match up with "number", which is singular. As a test, see how it scans with the prepositional phrase removed: "the number is limited" vs. "the number are limited".

Comment (lydia-duncan) [reply doc/rst/platforms/cray.rst:102]: D'oh, that's what happens when I respond to a single line instead of the whole sentence.  Sorry to bother you!

Comment (gbtitus) [reply doc/rst/platforms/cray.rst:102]: No probs -- in other circumstances that same thing would be valued, as: "has the ability to concentrate on the work at hand despite distractions."   :-)

**Pull request merged**